### PR TITLE
docs: fix python script bug

### PR DESCRIPTION
**Pull request by soulbird**
When there are only four parameters, the script should also work properly

### CODE REVIEW

**Comment by leslie-tsang** [comment]
Hello there, I'm afraid this change will not have the desired effect.

BTW, Could desc the PR more clearly ?  :)

**Comment by soulbird** [comment]
As the example, we can run this python script like: 
```shell
./patch_upstream_mtls.py testmtls ./client.pem ./client.key
```
it has four parameters, we expect it to work, but we will got 'bad argument'.

**Comment by soulbird** [comment]
> 



> Hello there, I'm afraid this change will not have the desired effect.
> 
> BTW, Could you desc the PR more clearly ? :)

As the example, we can run this python script like: 
```shell
./patch_upstream_mtls.py testmtls ./client.pem ./client.key
```
it has four parameters, we expect it to work, but we will got 'bad argument'.

**Comment by leslie-tsang** [comment]
> > 
> 
> > Hello there, I'm afraid this change will not have the desired effect.
> > BTW, Could you desc the PR more clearly ? :)
> 
> As the example, we can run this python script like:
> 
> ```shell
> ./patch_upstream_mtls.py testmtls ./client.pem ./client.key
> ```
> 
> it has four parameters, we expect it to work, but we will got 'bad argument'.

```python
if len(sys.argv) >= 5:
```

Can you confirm that this [conditional judgment](https://github.com/apache/apisix/blob/ce8227aaf7e4f0b8aab6274c38deda6e7c243c9d/docs/en/latest/mtls.md?plain=1#L121) continues to obstruct the request's execution?

**Comment by soulbird** [comment]
> > > 
> > 
> > 
> > > Hello there, I'm afraid this change will not have the desired effect.
> > > BTW, Could you desc the PR more clearly ? :)
> > 
> > 
> > As the example, we can run this python script like:
> > ```shell
> > ./patch_upstream_mtls.py testmtls ./client.pem ./client.key
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > it has four parameters, we expect it to work, but we will got 'bad argument'.
> 
> ```python
> if len(sys.argv) >= 5:
> ```
> 
> Can you confirm that this [conditional judgment](https://github.com/apache/apisix/blob/ce8227aaf7e4f0b8aab6274c38deda6e7c243c9d/docs/en/latest/mtls.md?plain=1#L121) continues to obstruct the request's execution?

Actuall, we expect it to work like:
```shell
./ssl.py ./server.pem ./server.key 'mtls.test.com'
```
or
```shell
./ssl.py ./server.pem ./server.key 'mtls.test.com' ./client_ca.pem 10
```

**Comment by leslie-tsang** [comment]
The change from `if len(sys.argv) <= 4` to `if len(sys.argv) < 4` in **<ins>ssl.py</ins>** don't have the desired effect.

There is another [conditional judgment](https://github.com/apache/apisix/blob/ce8227aaf7e4f0b8aab6274c38deda6e7c243c9d/docs/en/latest/mtls.md?plain=1#L121) `if len(sys.argv) >= 5` will prevent the request execution.

```patch
#!/usr/bin/env python
# coding: utf-8
# save this file as ssl.py
import sys
# sudo pip install requests
import requests
- if len(sys.argv) <= 4:
+ if len(sys.argv) < 4:
    print("bad argument")
    sys.exit(1)
with open(sys.argv[1]) as f:
    cert = f.read()
with open(sys.argv[2]) as f:
    key = f.read()
sni = sys.argv[3]
api_key = "edd1c9f034335f136f87ad84b625c8f1" # Change it
reqParam = {
    "cert": cert,
    "key": key,
    "snis": [sni],
}
if len(sys.argv) >= 5:
    print("Setting mTLS")
    reqParam["client"] = {}
    with open(sys.argv[4]) as f:
        clientCert = f.read()
        reqParam["client"]["ca"] = clientCert
    if len(sys.argv) >= 6:
        reqParam["client"]["depth"] = int(sys.argv[5])
resp = requests.put("http://127.0.0.1:9080/apisix/admin/ssl/1", json=reqParam, headers={
    "X-API-KEY": api_key,
})
print(resp.status_code)
print(resp.text)
```

**Comment by soulbird** [comment]
> 



> The change from `if len(sys.argv) <= 4` to `if len(sys.argv) < 4` in **ssl.py** don't have the desired effect.
> 
> There is another [conditional judgment](https://github.com/apache/apisix/blob/ce8227aaf7e4f0b8aab6274c38deda6e7c243c9d/docs/en/latest/mtls.md?plain=1#L121) `if len(sys.argv) >= 5` will prevent the request execution.
> 
> ```diff
> #!/usr/bin/env python
> # coding: utf-8
> # save this file as ssl.py
> import sys
> # sudo pip install requests
> import requests
> - if len(sys.argv) <= 4:
> + if len(sys.argv) < 4:
>     print("bad argument")
>     sys.exit(1)
> with open(sys.argv[1]) as f:
>     cert = f.read()
> with open(sys.argv[2]) as f:
>     key = f.read()
> sni = sys.argv[3]
> api_key = "edd1c9f034335f136f87ad84b625c8f1" # Change it
> reqParam = {
>     "cert": cert,
>     "key": key,
>     "snis": [sni],
> }
> if len(sys.argv) >= 5:
>     print("Setting mTLS")
>     reqParam["client"] = {}
>     with open(sys.argv[4]) as f:
>         clientCert = f.read()
>         reqParam["client"]["ca"] = clientCert
>     if len(sys.argv) >= 6:
>         reqParam["client"]["depth"] = int(sys.argv[5])
> resp = requests.put("http://127.0.0.1:9080/apisix/admin/ssl/1", json=reqParam, headers={
>     "X-API-KEY": api_key,
> })
> print(resp.status_code)
> print(resp.text)
> ```

It's no matter, it will work well with four or five or more parameters

**Comment by leslie-tsang** [comment]
Ohh, My mistake, we should add a blank line here like below:
```python
    if len(sys.argv) >= 6:
        reqParam["client"]["depth"] = int(sys.argv[5])

resp = requests.put("http://127.0.0.1:9080/apisix/admin/ssl/1", json=reqParam, headers={
    "X-API-KEY": api_key,
```

Thanks for your patience in replying. :)

LGTM

**Review by leslie-tsang — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

File: docs/en/latest/mtls.md [hunk 1]
```diff
@@ -103,7 +103,7 @@ import sys
 # sudo pip install requests
 import requests
 
-if len(sys.argv) <= 4:
+if len(sys.argv) < 4:
```
Comment (lingsamuel) [inline docs/en/latest/mtls.md:106]: I am afraid this is necessary.

Comment (bisakhmondal) [reply docs/en/latest/mtls.md:106]: It's okay, I think the script is being adapted to be used in storing server certs and keys in APISIX as well, along with mTLS client ca information. It's a minor change and if it ends up helping people, I think it's okay to be merged : )

File: docs/en/latest/mtls.md [hunk 2]
```diff
@@ -166,7 +166,7 @@ import sys
...
```
Comment (bisakhmondal) [inline docs/en/latest/mtls.md:169]: nice catch!

**Review by bisakhmondal — APPROVED** [review]

**Review by tzssangglass — APPROVED** [review]

**Pull request merged**